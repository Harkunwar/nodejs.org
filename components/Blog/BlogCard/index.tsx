import { FormattedMessage } from 'react-intl';
import styles from './index.module.scss';
import LocalizedLink from '../../LocalizedLink';
import navigation from '../../../navigation.json';
import type { BlogPost } from '../../../types';
import type { FC } from 'react';

const getBlogCategoryUrl = (category: string): string =>
  `${navigation.blog.link}/${category}/`;

const getBlogPostUrl = (slug: string) =>
  slug.endsWith('/') ? slug : `${slug}/`;

type Props = {} & Omit<BlogPost, 'file'>;

const BlogCard: FC<Props> = ({ title, author, date, category, slug }) => {
  const blogCategoryUrl = getBlogCategoryUrl(category);

  return (
    <div className={styles.blogCard}>
      <div className={styles.title}>
        <LocalizedLink href={getBlogPostUrl(slug)}>{title}</LocalizedLink>
        <div className={styles.metadata}>
          {category && (
            <LocalizedLink className={styles.category} href={blogCategoryUrl}>
              {category}
            </LocalizedLink>
          )}
        </div>
      </div>
      <div className={styles.content}>
        <h4>{date}</h4>
        {author && (
          <p>
            <FormattedMessage id="components.blog.blogCard.author.by" />{' '}
            <span>{author}</span>
          </p>
        )}
      </div>
    </div>
  );
};

export default BlogCard;
