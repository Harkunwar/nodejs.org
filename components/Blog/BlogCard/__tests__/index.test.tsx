import { render } from '@testing-library/react';
import { LocaleProvider } from '../../../../providers/localeProvider';
import englishMessages from '../../../../i18n/locales/en.json';
import BlogCard from '..';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({}),
}));

describe('BlogCard component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <LocaleProvider
        i18nData={{
          currentLocale: {
            code: 'en',
            localName: 'English',
            name: 'English',
            langDir: 'ltr',
            dateFormat: 'MM.DD.YYYY',
            hrefLang: 'en-US',
            enabled: true,
          },
          localeMessages: englishMessages,
        }}
      >
        <BlogCard
          author="Bat Man"
          category="category-mock"
          date="2023-04-21 23:40:56.77"
          slug="/blog/category-mock/sample-blog"
          title="Sample Test Blog"
        />
      </LocaleProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
