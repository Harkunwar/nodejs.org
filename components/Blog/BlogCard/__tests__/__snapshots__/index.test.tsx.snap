// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard component renders correctly 1`] = `
<div>
  <div
    class="blogCard"
  >
    <div
      class="title"
    >
      <a
        href="/en/blog/category-mock/sample-blog"
      >
        Sample Test Blog
      </a>
      <div
        class="metadata"
      >
        <a
          class="category"
          href="/en/blog/category-mock"
        >
          category-mock
        </a>
      </div>
    </div>
    <div
      class="content"
    >
      <h4>
        4/21/2023
      </h4>
      <p>
        by
         
        <span>
          Bat Man
        </span>
      </p>
    </div>
  </div>
</div>
`;
